Tasks/DeployVisualStudioTestAgent/SetupTestMachineForUITests.ps1

```diff
@@ -1,4 +1,115 @@
+		return $true
+	}
+	
+	$autoadminLogon = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).AutoAdminLogon
+	if ((!$autoadminLogon) -or ($autoadminLogon.Length -eq 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but AutoAdminLogon key is not set." -f $registryPath) -Verbose
+		return $true
+	}
+	elseif($autoadminLogon -eq "1")
+	{
+		return $false
+	}
+
+	Write-Verbose -Message("Registry path {0} found, but AutoAdminLogon key is not enabled." -f $registryPath) -Verbose
+	return $true
+}
+
+function IsShowLogonMessagePolicyEnabled([int] $OSType)
+{
+	#check 64 bit or 32 bit
+	if ($OSType -eq 64)
+	{
+		$registryPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\policies\system"
+	}
+	else
+	{
+		$registryPath = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\policies\system"
+	}
+
+	if (-not (Test-Path $registryPath))
+	{
+		Write-Verbose -Message("Registry path {0} not found" -f $registryPath) -Verbose
+		return $false
+	}
+	
+	$legalCaption = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).legalnoticecaption
+	$legalText = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).legalnoticetext
+	if (($legalCaption) -and ($legalCaption.Length -gt 0) -and ($legalText) -and ($legalText.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but legalnoticecaption and legaltext key is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	return $false
+}
+
+function IsShowLogonMessageEnabled()
+{
+    $registryPath = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon"
+
+	if (-not (Test-Path $registryPath))
+	{
+		Write-Verbose -Message("Registry path {0} not found" -f $registryPath) -Verbose
+		return $false
+	}
+	
+	$legalCaption = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).LegalNoticeCaption
+    $legalText = (Get-ItemProperty $registryPath -ErrorAction SilentlyContinue).LegalNoticeText
+	if (($legalCaption) -and ($legalCaption.Length -gt 0) -and ($legalText) -and ($legalText.Length -gt 0))
+	{
+		Write-Verbose -Message("Registry path {0} found, but LegalNoticeCaption and LegalNoticeText is set to some value." -f $registryPath) -Verbose
+		return $true
+	}
+
+	return $false
+}
+
+function Update-RebootCount([string] $environmentURL)
+{
+	if ([IntPtr]::Size -eq 8)
+	{
+		$testAgentRegPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\TestAgentConfig"
+	}
+	else
+	{
+		$testAgentRegPath = "HKLM:\SOFTWARE\Microsoft\TestAgentConfig" 
+	}
+
+	if (-not (Test-Path $testAgentRegPath))
+	{
+		New-Item -Path $testAgentRegPath -Force | Out-Null
+		Write-Verbose -Message ("Updating machine reboot count to 1") -Verbose
+		New-ItemProperty -Path $testAgentRegPath -Name "MachineRebootCount" -Value 1 -PropertyType DWord -Force | Out-Null
+		New-ItemProperty -Path $testAgentRegPath -Name "EnvironmentURL" -Value $environmentURL -PropertyType String -Force | Out-Null
```

Comment:
this doesn't need to be removed, as its renewed for every new environment url property